File: backend/src/domain/entities/sla_dashboard.py
# backend/src/domain/entities/sla_dashboard.py
from dataclasses import dataclass


@dataclass(frozen=True)
class SlaDashboardIssue:
    key: str
    type: str
    summary: str
    created: str
    updated: str
    status: str


@dataclass(frozen=True)
class SlaDashboardCommentImage:
    attachment_id: str
    alt: str


@dataclass(frozen=True)
class SlaDashboardComment:
    id: str
    author: str
    body: str
    created: str
    updated: str
    images: tuple[SlaDashboardCommentImage, ...]

File: frontend/src/domain/SlaDashboard.ts
// frontend/src/domain/SlaDashboard.ts
export interface SlaDashboardIssue {
  key: string
  type: string
  summary: string
  created: string
  updated: string
  status: string
}

export interface SlaDashboardComment {
  id: string
  author: string
  body: string
  created: string
  updated: string
  images: SlaDashboardCommentImage[]
}

export interface SlaDashboardCommentImage {
  attachment_id: string
  alt: string
}
